### PR TITLE
Ocular MISRA C Tutorial

### PR DESCRIPTION
**Pull request by NWeber-SL**
Source from Niko

### CODE REVIEW

File: using-ocular/tutorials/misra-c.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# Using ShiftLeft Ocular with MISRA C Applications
+
+[MISRA C](https://www.misra.org.uk/) is a set of software development guidelines for the C programming language developed by the Motor Industry Software Reliability Association (MISRA). Its aims are to facilitate code safety, security, portability and reliability in the context of embedded systems. You use ShiftLeft Ocular to examine the software elements and flows in your MISRA C applications to identify complex business logic vulnerabilities that can't be scanned for automatically.
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/tutorials/misra-c.md]:
```suggestion
[MISRA C](https://www.misra.org.uk/) is a set of software development guidelines for the C programming language developed by the Motor Industry Software Reliability Association (MISRA). Its aims are to facilitate code safety, security, portability and reliability in the context of embedded systems. You can use ShiftLeft Ocular to examine the software elements and flows in your MISRA C applications to identify complex business logic vulnerabilities that can't be scanned for automatically.
```

File: using-ocular/tutorials/misra-c.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+   fun1(arr);
+}
+```
+
+### ShiftLeft Ocular Query
+
+Use the following query to check for violations in the currently loaded Code Property Graph (CPG).
+
+
+```
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/tutorials/misra-c.md]:
```suggestion
```scala
```

File: using-ocular/tutorials/misra-c.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+of the MISRA 2012 standard.
+
+
+## Mandatory Rule 17.6 
+
+Rule 17.6 states that the declaration of an array parameter should not contain the `static` keyword between the []. This rule covers the possibility of developers assuming a fixed number of parameters provided to a function. Developers do so to increase performance, but with the risk that a function is called without the correct amount of parameters. 
+
+### Test Code
+
+```
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/tutorials/misra-c.md]:
```suggestion
```C
```

File: using-ocular/tutorials/misra-c.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+
+## Mandatory Rule 22.4 
+
+Rule 22.4 determines that in MISRA C, there should be no attempt to write to a stream which has been opened as read-only. Writing to a file that is only opened to read causes undefined behavior and thus should be avoided.
+
+### Test Code
+
+```
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/tutorials/misra-c.md]:
```suggestion
```C
```

File: using-ocular/tutorials/misra-c.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+void main() {
+    fn();
+}
+```
+
+### ShiftLeft Ocular Query
+
+Use the following query to check for violations in the currently loaded Code Property Graph (CPG).
+
+```
```
Comment (vlad-alexandru-ionescu) [inline using-ocular/tutorials/misra-c.md]:
```suggestion
```scala
```

File: using-ocular/tutorials/misra-c.md (hunk 1 above)
Comment (NWeber-SL) [reply using-ocular/tutorials/misra-c.md]: Done

File: using-ocular/tutorials/misra-c.md (hunk 2 above)
Comment (NWeber-SL) [reply using-ocular/tutorials/misra-c.md]: Added

File: using-ocular/tutorials/misra-c.md (hunk 3 above)
Comment (NWeber-SL) [reply using-ocular/tutorials/misra-c.md]: Added

File: using-ocular/tutorials/misra-c.md (hunk 4 above)
Comment (NWeber-SL) [reply using-ocular/tutorials/misra-c.md]: Added

File: using-ocular/tutorials/misra-c.md (hunk 5 above)
Comment (NWeber-SL) [reply using-ocular/tutorials/misra-c.md]: added

**Pull request merged**